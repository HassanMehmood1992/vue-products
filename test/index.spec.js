import { mount, createLocalVue } from "@vue/test-utils";
import { shallowMount } from "@vue/test-utils";
import Vuetify from "vuetify";
import Vuex from "vuex";

import Index from "@/pages/index";

const localVue = createLocalVue();
localVue.use(Vuex);

let wrapper;

beforeEach(() => {
  let vuetify = new Vuetify();

  wrapper = shallowMount(Index, {
    store: new Vuex.Store({
      state: { items: [], loading: false }
    }),
    localVue,
    vuetify,
    stubs: {
      NuxtLink: true,
      Nuxt: true
      // Any other component that you want stubbed
    }
  });
});

afterEach(() => {
  wrapper.destroy();
});

describe("Index/Index page", () => {
  test("is fully functional", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test("should have products title on the page", () => {
    const title = wrapper.find("#products");
    expect(title.element.innerHTML).toBe("Products");
  });

  test("should show error message if items are empty and not loading", () => {
    const wrapper = mount(Index, {
      data: () => ({
        items: [],
        loading: false
      }),
      computed: {
        reachedEnd: () => {
          return false;
        }
      }
    });
    const title = wrapper.find("#error-message");
    expect(title.element.innerHTML.length > 0).toBeTruthy();
    expect(title.element.innerHTML.length).toBeGreaterThan(0);
    wrapper.destroy()
  });
});
