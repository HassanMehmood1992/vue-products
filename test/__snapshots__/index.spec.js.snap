// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index/Index page is fully functional 1`] = `
<v-container-stub
  tag="div"
>
  <v-layout-stub
    class="my-10"
    row=""
    tag="div"
    wrap=""
  >
    <welcome-stub />
  </v-layout-stub>
   
  <v-layout-stub
    row=""
    tag="div"
    wrap=""
  >
    <v-flex-stub
      tag="div"
    >
      <v-card-stub
        class="mx-2"
        flat="true"
        loaderheight="4"
        tag="div"
      >
        <v-card-title-stub
          class="px-0"
          id="products"
        >
          Products
        </v-card-title-stub>
      </v-card-stub>
    </v-flex-stub>
  </v-layout-stub>
   
  <!---->
   
  <!---->
   
  <v-layout-stub
    class="mt-1"
    tag="div"
  >
    <v-flex-stub
      class="text-center"
      tag="div"
    >
      <v-progress-circular-stub
        color="primary"
        indeterminate="true"
        rotate="0"
        size="32"
        value="0"
        width="3"
      />
    </v-flex-stub>
  </v-layout-stub>
</v-container-stub>
`;
